docs/dump_javadump.md

```diff
@@ -27,8 +27,635 @@
+
+## LOCKS
+
+This section of the Java dump provides information about locks, which protect shared resources from being accessed by more than one entity at a time. The information is essential in a deadlock situation, where two threads attempt to synchronize on an object and lock an instance of a class. Precise information is recorded about the threads that are causing the problem, which enables you to identify the root cause.
+
+The following example shows a typical LOCKS section, where no deadlocks existed at the time the dump was triggered. For clarity, the following example shows a shortened version of this section, where `...` indicates that lines are removed:
+
+```
+NULL           ------------------------------------------------------------------------
+0SECTION       LOCKS subcomponent dump routine
+NULL           ===============================
+NULL           
+1LKPOOLINFO    Monitor pool info:
+2LKPOOLTOTAL     Current total number of monitors: 3
+NULL           
+1LKMONPOOLDUMP Monitor Pool Dump (flat & inflated object-monitors):
+2LKMONINUSE      sys_mon_t:0x00007FF4B0001D78 infl_mon_t: 0x00007FF4B0001DF8:
+3LKMONOBJECT       java/lang/ref/ReferenceQueue@0x00000000FFE26A10: <unowned>
+3LKNOTIFYQ            Waiting to be notified:
+3LKWAITNOTIFY            "Common-Cleaner" (J9VMThread:0x0000000000FD0100)
+NULL           
+1LKREGMONDUMP  JVM System Monitor Dump (registered monitors):
+2LKREGMON          Thread global lock (0x00007FF4F0004FE8): <unowned>
+2LKREGMON          &(PPG_mem_mem32_subAllocHeapMem32.monitor) lock (0x00007FF4F0005098): <unowned>
+2LKREGMON          NLS hash table lock (0x00007FF4F0005148): <unowned>
+...
+NULL           
+```
+
+## THREADS
+
+The THREADS section of a Java dump file provides summary information about the JVM thread pool and detailed information about Java threads, native threads, and stack traces. Understanding the content of this section can help you diagnose problems that are caused by blocked or waiting threads.
+
+A Java thread runs on a native thread. Several lines are recorded for each Java thread in the `Thread Details` subsection, which include the following key pieces of information:
+
+- `3XMTHREADINFO`: The thread name, address information for the VM thread structures and Java thread object, the thread state, and thread priority.
+- `3XMJAVALTHREAD`: The Java thread ID and daemon status from the thread object.
+- `3XMTHREADINFO1`: The native operating system thread ID, priority, scheduling policy, internal VM thread state, and VM thread flags.
+- `3XMTHREADINFO2`: The native stack address range.
+- `3XMTHREADINFO3`: Java callstack information (`4XESTACKTRACE`) or Native call stack information (`4XENATIVESTACK`).
+- `5XESTACKTRACE`: This line indicates whether locks were taken by a specific method.
+
+Java thread priorities are mapped to operating system priority values. The larger the thread priority, the more frequently the thread runs.
```

Comment:
> The larger the thread priority, the more frequently the thread runs.
Not necessarily true https://www.linuxjournal.com/node/10267

Reply:
OK, removed that statement.